src/logistration/RegistrationFailure.jsx

```diff
@@ -2,48 +2,61 @@ import React, { useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { FormattedMessage } from '@edx/frontend-platform/i18n';
 import { Alert } from '@edx/paragon';
+import { INTERNAL_SERVER_ERROR } from './data/constants';
 
-const hasNoErrors = (userErrors) => (
-  userErrors.every((errorList) => (!errorList[0]))
+const showAlert = (userErrors) => (
+  <Alert variant="danger">
+    <Alert.Heading className="text-danger">
+      <FormattedMessage
+        id="logistration.registration.request.failure.header.description.message"
+        defaultMessage="We couldn't create your account."
+        description="error message when registration failure."
+      />
+    </Alert.Heading>
+    <div>
+      <ul>
+        {userErrors}
+      </ul>
+    </div>
+  </Alert>
 );
 
 const RegistrationFailureMessage = (props) => {
   const errorMessage = props.errors;
+  const { errorCode } = props.errors;
   const userErrors = [];
 
   useEffect(() => {
     window.scrollTo({ left: 0, top: 0, behavior: 'smooth' });
   }, [props.submitCount]);
 
-  Object.keys(errorMessage).forEach((key) => {
-    const errors = errorMessage[key];
-    const errorList = errors.map((error) => (
-      (error.user_message) ? (
-        <li key={error} className="text-left">
-          {error.user_message}
+  let serverError;
+  switch (errorCode) {
+    case INTERNAL_SERVER_ERROR:
+      serverError = (
+        <li key={INTERNAL_SERVER_ERROR} className="text-left">
+          An error has occurred. Try refreshing the page, or check your Internet connection.
         </li>
-      ) : null
-    ));
-    userErrors.push(errorList);
-  });
+      );
+     userErrors.push(serverError);
+     break;
+
+     default:
+        Object.keys(errorMessage).forEach((key) => {
+          const errors = errorMessage[key];
+          const errorList = errors.map((error) => (
+            (error.user_message) ? (
+              <li key={error} className="text-left">
+                {error.user_message}
+              </li>
+            ) : null
+          ));
+          userErrors.push(errorList);
+        });
+  }
 
   return (
-    hasNoErrors(userErrors) ? null : (
-      <Alert variant="danger">
-        <Alert.Heading className="text-danger">
-          <FormattedMessage
-            id="logistration.registration.request.failure.header.description.message"
-            defaultMessage="We couldn't create your account."
-            description="error message when registration failure."
-          />
-        </Alert.Heading>
-        <div>
-          <ul>
-            {userErrors}
-          </ul>
-        </div>
-      </Alert>
-    )
+    !userErrors.length ? null : showAlert(userErrors)
```

Comment:
I think we keep the `<Alert>` elements in the return statement instead of creating a wrapper around it. Something like this can be done:
https://github.com/edx/frontend-app-logistration/blob/ad2ee9e013781618d9df01bf4e28217441562992/src/login/AccountActivationMessage.jsx#L50-L55

Reply:
+1 on this.